src/fparser/readfortran.py

```diff
@@ -1048,47 +1034,47 @@ def get_source_item(self):
-                    if self.isfix:     
+                    if self.isfix:
                         if i!=0:
                             message = message + ', switching to free format mode'
                         message = self.format_warning_message(\
                             message,startlineno, self.linecount)
-                        logger.warning(message)
-                        # self.show_message(message, sys.stderr)
+                        logging.getLogger(__name__).warning(message)
                         if i==0:
                             # non standard comment line:
-                            return self.comment_item(line, startlineno, startlineno)                           
+                            return self.comment_item(line, startlineno, startlineno)
                         self.set_mode(True, False)
                     else:
+                        message = self.format_warning_message(message,
+                                                              startlineno,
+                                                              self.linecount)
+                        logging.getLogger(__name__).warning(message)
                         if i==0:
-                            message = self.format_warning_message(\
-                            message,startlineno, self.linecount)
-                            logger.warning(message)
-                            # self.show_message(message, sys.stderr)
                             # non standard comment line:
-                            return self.comment_item(line, startlineno, startlineno)                           
+                            return self.comment_item(line, startlineno, startlineno)
                         # return line item with error message
                         # TODO: handle cases with line[6:]==''
                         return self.line_item(line[6:], startlineno, self.linecount,
                                            label, name, self.format_error_message(\
                             message, startlineno, self.linecount))
-            # check for label
-            s = line[:5].strip().lower()
-            if s:
-                label = int(s)
-            if not self.isf77:
-                m = _construct_name_re.match(line[6:])
-                if m:
-                    name = m.group('name')
-                    line = line[:6] + line[6:][m.end():].lstrip()
-            if not line[6:].strip():
-                # check for a blank line
-                if name is not None:
-                    self.error('No construct following construct-name.')
-                elif label is not None:
-                    self.warning('Label must follow nonblank character (F2008:3.2.5_2)')
-                return self.comment_item('', startlineno, self.linecount)
-            # line is not a comment and the start of the line is valid
+            if self.isfixed: # Check that we haven't switched to free format
```

Comment:
Yes. Previously it would carry on stripping the first 6 characters off the line and what have you even if previously it had swapped to free format mode. This could happen if you were in "fix" mode and the non-standard comment character was not the first in the line. I discovered this while working up the unit tests.